app/browser-detection.js

```diff
@@ -0,0 +1,226 @@
+        [ 'safari', /Version\/([0-9\._]+).*Safari/ ],
+        [ 'facebook', /FBAV\/([0-9\.]+)/],
+        [ 'instagram', /Instagram\ ([0-9\.]+)/],
+        [ 'ios-webview', /AppleWebKit\/([0-9\.]+).*Mobile/]
+      ]);
+    }
+
+    function getOperatingSystemRules() {
+      return buildRules([
+        [ 'iOS', /iP(hone|od|ad)/ ],
+        [ 'Android OS', /Android/ ],
+        [ 'BlackBerry OS', /BlackBerry|BB10/ ],
+        [ 'Windows Mobile', /IEMobile/ ],
+        [ 'Amazon OS', /Kindle/ ],
+        [ 'Windows 3.11', /Win16/ ],
+        [ 'Windows 95', /(Windows 95)|(Win95)|(Windows_95)/ ],
+        [ 'Windows 98', /(Windows 98)|(Win98)/ ],
+        [ 'Windows 2000', /(Windows NT 5.0)|(Windows 2000)/ ],
+        [ 'Windows XP', /(Windows NT 5.1)|(Windows XP)/ ],
+        [ 'Windows Server 2003', /(Windows NT 5.2)/ ],
+        [ 'Windows Vista', /(Windows NT 6.0)/ ],
+        [ 'Windows 7', /(Windows NT 6.1)/ ],
+        [ 'Windows 8', /(Windows NT 6.2)/ ],
+        [ 'Windows 8.1', /(Windows NT 6.3)/ ],
+        [ 'Windows 10', /(Windows NT 10.0)/ ],
+        [ 'Windows ME', /Windows ME/ ],
+        [ 'Open BSD', /OpenBSD/ ],
+        [ 'Sun OS', /SunOS/ ],
+        [ 'Linux', /(Linux)|(X11)/ ],
+        [ 'Mac OS', /(Mac_PowerPC)|(Macintosh)/ ],
+        [ 'QNX', /QNX/ ],
+        [ 'BeOS', /BeOS/ ],
+        [ 'OS/2', /OS\/2/ ],
+        [ 'Search Bot', /(nuhk)|(Googlebot)|(Yammybot)|(Openbot)|(Slurp)|(MSNBot)|(Ask Jeeves\/Teoma)|(ia_archiver)/ ]
+      ]);
+    }
+
+    function buildRules(ruleTuples) {
+      return ruleTuples.map(function(tuple) {
+        return {
+          name: tuple[0],
+          rule: tuple[1]
+        };
+      });
+    }
+
+    function renderNotSupported(){
+        var cqNotSupported = document.createElement('div');
+        cqNotSupported.className = 'cq-not-supported';
+
+        var cqLogo = document.createElement('div');
+        cqLogo.className = 'cq-logo';
+        cqNotSupported.appendChild(cqLogo);
+
+        var cqIconNotSupported = document.createElement('div');
+        cqIconNotSupported.className = 'cq-icon-not-supported';
+        cqNotSupported.appendChild(cqIconNotSupported);
+
+        var cqH1 = document.createElement('h1');
+        cqH1.innerHTML = 'YOUR BROWSER IS NOT SUPPORTED';
+        cqNotSupported.appendChild(cqH1);
+
+        var cqP = document.createElement('p');
+        cqP.innerHTML = 'Please update your browser for the best trading experience.';
+        cqNotSupported.appendChild(cqP);
+
+        var cqA = document.createElement('a');
+        cqA.href = 'http://outdatedbrowser.com/';
+        cqA.innerHTML = 'Update Browser';
+        cqNotSupported.appendChild(cqA);
+        
+        if (cqRoot.nextSibling) {
+            cqRoot.parentNode.insertBefore(cqNotSupported, cqRoot.nextSibling);
+        } else {
+            cqRoot.parentNode.appendChild(cqNotSupported);
+        }
+        cqRoot.parentNode.removeChild(cqRoot);
```

Comment:
sometimes we show `not supported` page because of CSS or any critical issue, so as I didn't find any safe way to warranty stop running rest of codes, I decide to stop it by removing `<div id="root" />`.